src/main/kotlin/org/rust/ide/typing/RsAngleBraceHandlers.kt

```diff
@@ -0,0 +1,136 @@
+package org.rust.ide.typing
+
+import com.intellij.codeInsight.CodeInsightSettings
+import com.intellij.codeInsight.editorActions.TypedHandlerDelegate
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.tokenSetOf
+
+private val GENERIC_NAMED_ENTITY_KEYWORDS = tokenSetOf(RsElementTypes.FN, RsElementTypes.STRUCT,
+    RsElementTypes.ENUM, RsElementTypes.TRAIT, RsElementTypes.TYPE_KW)
+
+private val INVALID_INSIDE_TOKENS = tokenSetOf(RsElementTypes.LBRACE, RsElementTypes.RBRACE,
+    RsElementTypes.SEMICOLON)
+
+class RsAngleBraceTypedHandler : TypedHandlerDelegate() {
+
+    private var rsLTTyped = false
+
+    override fun beforeCharTyped(c: Char, project: Project, editor: Editor, file: PsiFile, fileType: FileType): Result {
+        if (file !is RsFile) {
+            return Result.CONTINUE
+        }
```

Comment:
It's Ok to write simple  "bail out" ifs on one line: `if (file !is RsFile) return Result.CONTINUE`.